### ISSUE TITLE
Issue Generating Underground Structure 20w30a

### ISSUE THREAD

**Issue by TheBrokenRail** [Earthcomputer/LibStructure#2]
```java
        Registry.register(Registry.STRUCTURE_PIECE, new Identifier(EnergonRelics.NAMESPACE, name + "_piece"), piece);
        LibStructure.registerStructure(new Identifier(EnergonRelics.NAMESPACE, name), feature, GenerationStep.Feature.UNDERGROUND_STRUCTURES, structureConfig, feature.configure(FeatureConfig.DEFAULT));
```
is ran in ``onInitialize``.

The structure is added to biomes with a mixin to ``DefaltBiomeFeatures``:
```java
@Mixin(DefaultBiomeFeatures.class)
public class MixinDefaultBiomeFeatures {
    @Inject(at = @At("RETURN"), method = "addDefaultUndergroundStructures")
    private static void addDefaultUndergroundStructures(Biome biome, CallbackInfo info) {
        biome.addStructureFeature(entry.getValue().feature.configure(DefaultFeatureConfig.INSTANCE));
    }
}
```

This crashes with:
```
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:194)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:140)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:192)
	... 3 more
Caused by: java.lang.ExceptionInInitializerError
	at net.minecraft.util.registry.BuiltinRegistries.method_30567(BuiltinRegistries.java:52)
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries.java:75)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries.java:74)
	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:238)
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:42)
	at net.minecraft.client.main.Main.main(Main.java:137)
	... 8 more
Caused by: java.lang.NullPointerException
	at net.minecraft.world.biome.Biome.addStructureFeature(Biome.java:510)
	at com.thebrokenrail.energonrelics.block.structure.StructureGeneratorBlock.addToBiome(StructureGeneratorBlock.java:123)
	at net.minecraft.world.gen.feature.DefaultBiomeFeatures.handler$zcb000$addDefaultUndergroundStructures(DefaultBiomeFeatures.java:522)
	at net.minecraft.world.gen.feature.DefaultBiomeFeatures.addDefaultUndergroundStructures(DefaultBiomeFeatures.java:17)
	at net.minecraft.world.biome.DefaultBiomeCreator.createPlains(DefaultBiomeCreator.java:372)
	at net.minecraft.world.biome.Biomes.<clinit>(Biomes.java:17)
	... 15 more
```
because the mixin is used before ``onItialize`` is called, and ``feature.getGenerationStep`` uses the registry as it is not registered yet, it is ``null``.

Howver if I register the structure in a ``static {}`` block the error is:
```
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:194)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:140)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:192)
	... 3 more
Caused by: java.lang.ExceptionInInitializerError
	at net.earthcomputer.libstructure.LibStructure.registerStructure(LibStructure.java:33)
	at com.thebrokenrail.energonrelics.block.structure.StructureGeneratorBlock.register(StructureGeneratorBlock.java:107)
	at com.thebrokenrail.energonrelics.block.structure.StructureGeneratorBlock.register(StructureGeneratorBlock.java:117)
	at com.thebrokenrail.energonrelics.block.structure.StructureGeneratorBlock.<clinit>(StructureGeneratorBlock.java:129)
	at net.minecraft.world.gen.feature.DefaultBiomeFeatures.handler$zcb000$addDefaultUndergroundStructures(DefaultBiomeFeatures.java:522)
	at net.minecraft.world.gen.feature.DefaultBiomeFeatures.addDefaultUndergroundStructures(DefaultBiomeFeatures.java:17)
	at net.minecraft.world.biome.DefaultBiomeCreator.createPlains(DefaultBiomeCreator.java:372)
	at net.minecraft.world.biome.Biomes.<clinit>(Biomes.java:17)
	at net.minecraft.util.registry.BuiltinRegistries.method_30567(BuiltinRegistries.java:52)
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries.java:75)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries.java:74)
	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:238)
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:42)
	at net.minecraft.client.main.Main.main(Main.java:137)
	... 8 more
Caused by: java.lang.IllegalStateException: Structure feature without default settings: energonrelics:research_complex
	at net.minecraft.world.gen.chunk.StructuresConfig.<clinit>(StructuresConfig.java:56)
	... 23 more
```

Also, the following is printed to the log on startup:
```
[12:31:29] [main/WARN] (mixin) @Mixin target net.minecraft.class_5284$class_5307 was not found libstructure.mixins.json:ChunkGeneratorTypePresetMixin
```

**Comment by Earthcomputer** [comment]
Can you link me the repository which is causing this crash? Your mixin doesn't compile.

**Comment by TheBrokenRail** [comment]
Here you go: https://gitea.thebrokenrail.com/TheBrokenRail/EnergonRelics

**Comment by Earthcomputer** [comment]
Try the second approach with LibStructure 1.4

**Comment by TheBrokenRail** [comment]
It is still broken with v1.4:
```
[14:30:19] [main/INFO] (FabricLoader) Loading for game Minecraft 20w30a
[14:30:20] [main/INFO] (Fabric|Loader) [FabricLoader] Loading 46 mods: fabric-networking-blockentity-v0@0.2.5+b50ffc7bd3, minecraft@1.16.2-alpha.20.30.a, fabric-screen-handler-api-v1@1.1.0+87249841d3, fabric-renderer-api-v1@0.2.13+eae12eb8d3, fabric-keybindings-v0@0.2.0+3fa9f7c5d3, modmenu@1.14.5+build.30, fabric-command-api-v1@1.0.8+5ce53398d3, cloth-basic-math@0.5.1, fabricloader@0.9.0+build.204, energonrelics@0.0.1, fabric-tool-attribute-api-v1@1.2.0+d21d4635d3, fabric-renderer-indigo@0.3.4+5d32f583d3, fabric-containers-v0@0.1.8+045df74fd3, fabric-object-builder-api-v1@1.5.6+ba4afa57d3, fabric-biomes-v1@1.0.2+d3726004d3, fabric-item-api-v1@1.0.0+16acbe5bd3, fabric-crash-report-info-v1@0.1.2+b7f9825dd3, fabric-events-interaction-v0@0.3.3+7066030fd3, fabric-game-rule-api-v1@1.0.0+deaf4831d3, fabric-api-base@0.1.3+12a8474cd3, fabric-rendering-v0@1.1.0+53410490d3, fabric-key-binding-api-v1@1.0.1+730711c6d3, fabric-rendering-v1@1.1.2+346247d7d3, cloth-config2@4.6.0, fabric-rendering-data-attachment-v1@0.1.3+b7f9825dd3, autoconfig1u@3.2.0-unstable, fabric-resource-loader-v0@0.2.8+35e08e33d3, fabric-textures-v0@1.0.4+eae12eb8d3, fabric-content-registries-v0@0.1.9+059ea866d3, fabric-tag-extensions-v0@1.0.1+35e08e33d3, fabric-rendering-fluids-v1@0.1.8+b7084faad3, fabric-commands-v0@0.2.0+52d30836d3, fabric-registry-sync-v0@0.3.10+5553fa58d3, fabric-blockrenderlayer-v1@1.1.4+c6a8ea89d3, fabric-lifecycle-events-v1@1.1.0+3b4b2c13d3, fabric-mining-levels-v0@0.1.2+b764ce99d3, fabric-renderer-registries-v1@2.0.1+5a0f9a60d3, fabric-loot-tables-v1@1.0.0+fac8f366d3, fabric-events-lifecycle-v0@0.2.0+16acbe5bd3, fabric@0.16.0+build.386-1.16, libstructure@1.4, fabric-item-groups-v0@0.2.0+438f9636d3, fabric-particles-v1@0.2.1+0a6f2a70d3, fabric-models-v0@0.1.0+dfdb52d6d3, fabric-networking-v0@0.1.10+e00ecb5fd3, fabric-object-builders-v0@0.6.1+ba4afa57d3
[14:30:21] [main/INFO] (mixin) SpongePowered MIXIN Subsystem Version=0.8 Source=file:/home/connor/.gradle/caches/modules-2/files-2.1/net.fabricmc/sponge-mixin/0.8+build.18/af077ad2845a00cde186fa2c99c4dd32bd2a2766/sponge-mixin-0.8+build.18.jar Service=Knot/Fabric Env=CLIENT
[14:30:21] [main/INFO] (Fabric|MixinBootstrap) Loaded Fabric development mappings for mixin remapper!
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-networking-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-networking-blockentity-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-biomes-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-screen-handler-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-lifecycle-events-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-containers-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-renderer-indigo-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-rendering-data-attachment-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-item-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-events-interaction-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-loot-tables-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-object-builder-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-renderer-registries-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-textures-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-registry-sync-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-models-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-blockrenderlayer-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-renderer-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-containers-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-rendering-fluids-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-resource-loader-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-game-rule-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-particles-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap libstructure-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-object-builders-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-tag-extensions-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-tool-attribute-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-rendering-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-item-groups-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-content-registries-v0-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-command-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-crash-report-info-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-key-binding-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap fabric-renderer-api-v1-refmap.json using remapper chain
[14:30:22] [main/INFO] (mixin) Remapping refMap modmenu.refmap.json using remapper chain
[14:30:22] [main/WARN] (mixin) @Mixin target net/minecraft/server/world/ThreadedAnvilChunkStorage$EntityTracker is public in fabric-networking-v0.mixins.json:MixinEntityTracker and should be specified in value
[14:30:22] [main/WARN] (mixin) @Mixin target net/minecraft/client/render/block/BlockModelRenderer$AmbientOcclusionCalculator is public in fabric-renderer-indigo.mixins.json:MixinAmbientOcclusionCalculator and should be specified in value
[14:30:22] [main/WARN] (mixin) @Mixin target net/minecraft/client/render/chunk/ChunkBuilder$BuiltChunk$RebuildTask is public in fabric-renderer-indigo.mixins.json:MixinChunkRebuildTask and should be specified in value
[14:30:23] [main/WARN] (mixin) @Mixin target net/minecraft/server/command/GameRuleCommand$1 is public in fabric-game-rule-api-v1.mixins.json:GameRuleCommandVisitorMixin and should be specified in value
[14:30:23] [main/WARN] (mixin) @Mixin target net/minecraft/client/gui/screen/world/EditGameRulesScreen$RuleListWidget$1 is public in fabric-game-rule-api-v1.mixins.json:client.RuleListWidgetVisitorMixin and should be specified in value
[14:30:23] [main/WARN] (mixin) @Mixin target net/minecraft/client/particle/ParticleManager$SimpleSpriteProvider is public in fabric-particles-v1.mixins.json:ParticleManagerAccessor$SimpleSpriteProviderAccessor and should be specified in value
[14:30:23] [main/WARN] (mixin) @Mixin target net/minecraft/tag/Tag$1 is public in fabric-tag-extensions-v0.mixins.json:MixinTagImpl and should be specified in value
[14:30:23] [main/WARN] (mixin) Error loading class: net/minecraft/tag/GlobalTagAccessor$CachedTag (java.lang.ClassNotFoundException: net/minecraft/tag/GlobalTagAccessor$CachedTag)
[14:30:23] [main/WARN] (mixin) @Mixin target net/minecraft/tag/GlobalTagAccessor$CachedTag was not found fabric-tag-extensions-v0.mixins.json:MixinTagImpl
[14:30:23] [main/WARN] (mixin) @Mixin target net/minecraft/client/gui/widget/EntryListWidget$Entries is public in mixins.modmenu.json:MixinEntryListWidgetEntries and should be specified in value
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:194)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:140)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:192)
	... 3 more
Caused by: java.lang.ExceptionInInitializerError
	at net.earthcomputer.libstructure.LibStructure.registerStructure(LibStructure.java:36)
	at com.thebrokenrail.energonrelics.block.structure.StructureGeneratorBlock.create(StructureGeneratorBlock.java:114)
	at com.thebrokenrail.energonrelics.block.structure.StructureGeneratorBlock.<clinit>(StructureGeneratorBlock.java:130)
	at net.minecraft.world.gen.feature.DefaultBiomeFeatures.handler$zcb000$addDefaultUndergroundStructures(DefaultBiomeFeatures.java:522)
	at net.minecraft.world.gen.feature.DefaultBiomeFeatures.addDefaultUndergroundStructures(DefaultBiomeFeatures.java:17)
	at net.minecraft.world.biome.DefaultBiomeCreator.createPlains(DefaultBiomeCreator.java:372)
	at net.minecraft.world.biome.Biomes.<clinit>(Biomes.java:17)
	at net.minecraft.util.registry.BuiltinRegistries.method_30567(BuiltinRegistries.java:52)
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries.java:75)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries.java:74)
	at net.minecraft.util.registry.Registry.<clinit>(Registry.java:238)
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:42)
	at net.minecraft.client.main.Main.main(Main.java:137)
	... 8 more
Caused by: java.lang.IllegalStateException: Structure feature without default settings: energonrelics:research_complex
	at net.minecraft.world.gen.chunk.StructuresConfig.<clinit>(StructuresConfig.java:56)
	... 22 more
```

**Comment by Earthcomputer** [comment]
Fixed in 1.4.1
